File: 2.Algorithm/99.ETC/키패드 누르기/키패드 누르기.py
def solution(numbers, hand):
    answer = ""
    left = [3,1]
    right = [3,3]
    hand = -0.1 if hand == "left" else 0.1
    for i in numbers:
        i = i if i >0 else 11
        if i % 3 == 1:
            answer += "L"
            left = [i //3 , 1]
        elif i % 3 == 0:
            answer += "R"
            right = [i//3-1, 3]
        else:
            distanceLeft = abs(left[0]-(i // 3)) + abs(left[1]-(i % 3)) + hand
            distanceRight = abs(right[0]-(i // 3)) + abs(right[1]-(i % 3))
            if distanceLeft < distanceRight:
                answer += "L"
                left = [i//3,2]
            else:
                answer += "R"
                right = [i//3,2]
    return answer

File: 2.Algorithm/77.Kakao/키패드 누르기(Kakao)/키패드 누르기.java
class Solution {
    public String solution(int[] numbers, String hand) {
        String answer = "";
        int[] left = {3,1};
        int[] right = {3,3};
        double handadd = (hand.equals("left") ? -0.1 : 0.1);
        for (int i : numbers){
            i = (i > 0 ? i : 11);
            if (i % 3 == 1){
                answer += "L";
                left = new int[] {i/3, 1}; 
            }else if (i % 3 == 0){
                answer += "R";
                right = new int[] {i/3-1,3};
            }else{
                double distanceLeft = Math.abs(left[0] - (i/3)) + 
                    Math.abs(left[1]-(i%3)) + handadd;
                int distanceRight = Math.abs(right[0] - (i/3)) + 
                    Math.abs(right[1]-(i%3));
                if (distanceLeft < distanceRight){
                    answer += "L";
                    left = new int[] {i/3, 2}; 
                }else {
                    answer += "R";
                    right = new int[] {i/3,2};
                }
            }
        }
        return answer;
    }
}